src/functionality.py

```diff
@@ -49,24 +150,652 @@ def _validate_command(ctx: commands.Context) -> bool:
+
+            if not profile:
+                return await ctx.send("`That Serial Code does not exist in the database.`")
+
+            for rp in profile:
+                date_ = time.strftime('%d-%B-%Y %H:%M:%S', time.gmtime(int(rp["rp_start_date"])))
+                embed = discord.Embed(title=f'RP Profile : {rp["rp_name"]}',
+                                      description=f'**RP Name**: {rp["rp_name"]}\n'
+                                                  f'**Serial Code**: {rp["serial_code"]}\n'
+                                                  f'**Approved by**: {rp["approved"]}\n'
+                                                  f'**Main Host**: {rp["main_host"]}\n'
+                                                  f'**RP First Session Date**: {date_}\n'
+                                                  f'**RP Start Time**: {rp["rp_start_time"]}\n'
+                                                  f'**RP Length in Hours**: {rp["rp_duration"]}\n'
+                                                  f'**Local DRO**: {bool(int(rp["local"]))}\n'
+                                                  f'**Sign Up Open**: {bool(int(rp["sign_up"]))}\n'
+                                                  f'**Ongoing**: {bool(int(rp["ongoing"]))}\n'
+                                                  f'**Ended**: {bool(int(rp["ended"]))}\n'
+                                                  f'**Document**: {rp["doc"]}',
+
+                                      colour=discord.Color.dark_blue())
+                embed.set_thumbnail(url=ctx.author.avatar_url)
+                embed.set_footer(text=ctx.author)
+
+            return await ctx.send(embed=embed)
+
+        @bot.command(
+            name='rp_profile_filter',
+            brief='Returns a list of RP profiles that are filtered.',
+            help=('Returns a list of RP profiles that are filtered based on value of 0-3'
+                  '\nArguments: $rp_profile_filter <value>'
+                  '\nExample: $rp_profile_filter 0'
+                  '\n<value> is the code for filter action.'
+                  '\n0 - Filters RP profiles whose Sign Ups are Open.'
+                  '\n1 - Filters RP profiles which are currently Ongoing.'
+                  '\n2 - Filters RP Profiles which have ended.'
+                  '\n3 - Does not filter and returns all RP profiles.')
+        )
+        async def rp_profile_filter(ctx: commands.Context, value: int = 0):
+            if not _validate_command(ctx):
+                return
+
+            try:
+                value = int(value)
+            except ValueError:
+                return await ctx.send("`Input a valid value from 0-3 only.`")
+
+            rp_dict = {
+                0: rp_profile_check_sign_up(value),
+                1: rp_profile_check_ongoing(value),
+                2: rp_profile_check_ended(value),
+                3: rp_profile_check_all(value)
+            }
+
+            profile = rp_dict[value]
```

Comment:
This raises a silent KeyError if the user puts in some value other than numbers in the set `{0, 1, 2, 3}`. Please add an explicit error message.